CSharp/Common/GettingStartedCommon.cs

```diff
@@ -59,7 +59,7 @@ public static class GettingStartedCommon
         public static string CreateJobId(string prefix)
         {
             // a job is uniquely identified by its ID so your account name along with a timestamp is added as suffix
-            return string.Format("{0}-{1}-{2}", prefix, Environment.GetEnvironmentVariable("USERNAME"), DateTime.Now.ToString("yyyyMMdd-HHmmss"));
+            return string.Format("{0}-{1}-{2}", prefix, Environment.UserName.Replace(" ", ""), DateTime.Now.ToString("yyyyMMdd-HHmmss"));
```

Comment:
Good catch!
While you're modifying this code, would you mind updating it to something like this instead?

string formattedUserName = new string(Environment.UserName.Where(c => Char.IsLetterOrDigit(c)).ToArray());
return string.Format("{0}-{1}-{2}", prefix, formattedUserName, DateTime.Now.ToString("yyyyMMdd-HHmmss"));

That should be a more comprehensive fix to remove valid username chars that can cause problems with the Batch Service.